.github/workflows/ci.yaml

```diff
@@ -0,0 +1,108 @@
+name: CI Pipeline
+
+on:
+  workflow_call:
+    secrets:
+      MAIL_USERNAME:
+        description: 'Username for the email notification step'
+        required: true
+      MAIL_PASSWORD:
+        description: 'Password for the email notification step'
+        required: true
+    inputs:
+      custom_docker_tag:
+        description: 'Custom Docker tag'
+        # If not specified, then the 'latest' tag will be used
+        default: ''
+        required: false
+        type: string
+
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  DOCKER_REGISTRY: ghcr.io
+  DOCKER_REPO: angular-react-starter-image
+  DOCKER_TAG: ${{ inputs.custom_docker_tag || 'latest' }}
+  MAIL_SERVER_ADDRESS: smtp.gmail.com
+  MAIL_SERVER_PORT: 465
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./angular
+    steps:
+      - name: Checkout code from the repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Install Clean dependencies
+        run: npm ci
+
+      - name: Run headless tests
+        id: run-tests
+        if: ${{ always() }}
+        run: npm run test -- --watch=false --browsers=ChromeHeadless 
+
+      - name: Build Angular project
+        id: build-project
+        if: steps.run-tests.outcome == 'success'
+        run: npm run build
+
+      - name: Log in to GitHub Container Registry  
+        id: login-registry
```

Comment:
have a consistent naming
- Camelcase some other words
- Camelcase Some Other Words

but don't mix. This is like sore in the eye as well.